lvm_tests: tests should not assume system configuration

### Description

LVM tests try to fake a config using lvm filter, but this cannot work for testing real lvm commands. The test must not assume anything about the system configuration.

This command is wrong:
```python
    # Simulate removal of the second PV on another host, leaving stale PV in
    # the cache.
    commands.run([
        "vgreduce",
        "--config", tmp_storage.lvm_config(),
        vg_name,
        stale_pv_name,
    ])

```

The test and the code can also not change the system configuration, it should adapt to it and use `--devices` on this system.

### Reproduce

Run a single test that employs the `stale_pv` fixture as root on a CentOS stream 8 or other supported system, e.g.:
- `sudo ~/.venv/vdsm/bin/tox -e storage tests/storage/lvm_test.py -- -k test_pv_stale_reload_one_stale`

Test will fail with:
```
E           vdsm.common.cmdutils.Error: Command ['vgreduce', '--config', 'global { use_lvmetad=0 } devices { write_cache_state=0 filter=["a|^/dev/loop10$|^/dev/loop11$|", "r|.*|"] }', '5f6c5d91-b734-4785-99a8-13aa137f61c0', '/dev/loop11'] failed with rc=5 out=b'' err=b'  Cannot use /dev/loop11: device is not in devices file\n  Volume group "5f6c5d91-b734-4785-99a8-13aa137f61c0" not found\n  Cannot process volume group 5f6c5d91-b734-4785-99a8-13aa137f61c0\n'
```

### Actions to fix it

We can fix this in two steps:
1. Check how we build the command in the real code, and replace the manual code
2. Ideally, we should really run these tests in a new VM started for this purpose, to allow quick feedback

--- a/tests/storage/lvm_test.py
+++ b/tests/storage/lvm_test.py
@@ -1937,7 +1937,7 @@ def stale_lv(tmp_storage):
     commands.run([
         "lvremove",
         "--devices", dev,
-        "{}/{}".format(vg_name, stale_lv_name),
+        f"{vg_name}/{stale_lv_name}",
     ])
 
     # The cache still keeps both lvs.

--- a/tests/storage/lvm_test.py
+++ b/tests/storage/lvm_test.py
@@ -1627,6 +1627,7 @@ def test_lv_activate_deactivate(tmp_storage):
 @requires_root
 @pytest.mark.root
 def test_lv_deactivate_in_use(tmp_storage):
+    # TODO: This test does not work in active hosts.
     dev_size = 1 * GiB
     dev = tmp_storage.create_device(dev_size)
     vg_name = str(uuid.uuid4())

--- a/tests/storage/lvm_test.py
+++ b/tests/storage/lvm_test.py
@@ -1038,17 +1038,18 @@ def stale_pv(tmp_storage):
     pvs = sorted(pv.name for pv in lvm.getAllPVs())
     assert pvs == sorted([good_pv_name, stale_pv_name])
 
+    devices = ",".join([good_pv_name, stale_pv_name])
     # Simulate removal of the second PV on another host, leaving stale PV in
     # the cache.
     commands.run([
         "vgreduce",
-        "--config", tmp_storage.lvm_config(),
+        "--devices", devices,
         vg_name,
         stale_pv_name,
     ])
     commands.run([
         "pvremove",
-        "--config", tmp_storage.lvm_config(),
+        "--devices", devices,
         stale_pv_name,
     ])
 
@@ -1513,7 +1514,7 @@ def stale_vg(tmp_storage):
     # the cache.
     commands.run([
         "vgremove",
-        "--config", tmp_storage.lvm_config(),
+        "--devices", dev2,
         stale_vg_name,
     ])
 
@@ -1753,7 +1754,7 @@ def test_lv_refresh(tmp_storage):
     # Simulate extending the LV on the SPM.
     commands.run([
         "lvextend",
-        "--config", tmp_storage.lvm_config(),
+        "--devices", dev,
         "-L+1g",
         lv_fullname
     ])
@@ -1766,7 +1767,7 @@ def test_lv_refresh(tmp_storage):
     # Simulate extending the LV on the SPM.
     commands.run([
         "lvextend",
-        "--config", tmp_storage.lvm_config(),
+        "--devices", dev,
         "-L+1g",
         lv_fullname
     ])
@@ -1935,7 +1936,7 @@ def stale_lv(tmp_storage):
     # the cache.
     commands.run([
         "lvremove",
-        "--config", tmp_storage.lvm_config(),
+        "--devices", dev,
         "{}/{}".format(vg_name, stale_lv_name),
     ])
 

